core/rpc/core/src/impl/utils.rs

```diff
@@ -851,12 +851,20 @@ impl<C: CkbRpc> MercuryRpcImpl<C> {
             .occupied_capacity(data_occupied)
             .map_err(|e| CoreError::OccupiedCapacityError(e.to_string()))?;
 
+        let mut occupied = occupied.as_u64();
         // To make CKB `free` represent available balance, pure ckb cell should be spendable.
-        let occupied = if cell.cell_data.is_empty() && cell.cell_output.type_().is_none() {
-            0
-        } else {
-            occupied.as_u64()
-        };
+        if cell.cell_data.is_empty() && cell.cell_output.type_().is_none() {
+            occupied = 0;
+        }
+        // sUDT cell with 0 udt amount should be spendable.
```

Comment:
For 0 udt amount, all its capacity is spendable. For non-zero amount, its capacity minus occupied should be considered as spendable.